packages/commonwealth/client/scripts/helpers/chain.ts

```diff
@@ -43,7 +43,10 @@ export const selectChain = async (chain?: ChainInfo): Promise<boolean> => {
   }
 
   // Shut down old chain if applicable
-  await deinitChainOrCommunity();
+  if (!app.skipDeinitChain) {
```

Comment:
Upon further investigation, deinitChainOrCommuntiy no longer ran on community navigation at all. As a result I had re-added it in, and added in this bandaid fix. Its a simple global boolean variable that I will toggle on when I update the admin role.

I am aware this is bad practice, but If there are any better ideas, I would happily change it. This is just the only way I can think of that won't touch everything and have consequences that are difficult to reason with.

Reply:
@kurtisassad this solution is fine for the time being, but can you add a comment explaining the hack involved, so we can make a note to remove it later when we refactor the initialization flow further?